app/src/main/java/com/github/godspeed010/weblib/fragments/WebViewFragment.kt

```diff
@@ -93,10 +101,60 @@ class WebViewFragment : Fragment() {
                     view.findViewById<EditText>(R.id.et_address_bar).setText(url)
 
                     if (url != null) {
+                        pageError = false
+                        lastProgression = 0f
                         lastVisitedUrl = url
                     }
                 }
             }
+
+            override fun onReceivedHttpError(wv: WebView?, request: WebResourceRequest?, errorResponse: WebResourceResponse?) {
+                super.onReceivedHttpError(wv, request, errorResponse)
+
+                if (request?.url.toString() == wv?.url) pageError = true
+            }
+
+            override fun onPageFinished(wv: WebView?, loadedUrl: String?) {
+                super.onPageFinished(wv, loadedUrl)
+
+                if ((loadedUrl == novel.url) && !pageError) {
+                    if (novel.progression != 0f) {
+                        val toast: Toast = Toast.makeText(context, "Please wait for page to scroll.", Toast.LENGTH_LONG)
+                        toast.show()
+                        wv?.postDelayed({
+                            val scrollY: Int = calculateScrollYFromProgression(novel.progression, wv)
+                            val progressionPct: String = NumberFormat.getPercentInstance().let {
+                                it.minimumFractionDigits = 1
+                                it.format(novel.progression)
+                            }
+                            Log.d(
+                                TAG,
+                                "Page finished loading, scrolling to $scrollY ($progressionPct)"
+                            )
+                            wv.scrollTo(0, scrollY)
+                            toast.cancel()
+                        }, 2000)
+                    }
+                }
+
+                lastPageTitle = wv?.title ?: loadedUrl!!
```

Comment:
Good points, I will look into updating this.

Reply:
I think I did plan on changing so instead of showing the URL of the last chapter read it'd show the title. Consequently I had chosen to use the URL as a fallback since it would be more up to date than the previous title.

Reply:
Also looking at the code some more, novel.title would currently refer to the user-defined name for the WebNovel, so I think at this time, I'll just remove the lastPageTitle variable since it's not being used, and will work on it as a future feature.